src/Stratis.Tests.Common/Stratis.Tests.Common.csproj

```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="FluentAssertions" Version="5.2.0" />
+    <PackageReference Include="Moq" Version="4.8.2" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Stratis.Bitcoin.Features.Wallet\Stratis.Bitcoin.Features.Wallet.csproj" />
```

Comment:
The wallet helpers class is used by a few wallet projects but it's fine if it's in side the wallet.tests project.
Having a reference to the wallet feature defeats the purpose I think.
If we ever want to have Tests.Common as a nuget, it'll be dependant on many other packages.

Reply:
Perhaps yes the wallet tests helper class in here is a step too far. I did initially duplicate the re-used code to break the dependency between unit tests. As only a small part of it was actually being used and the majority was used in the main wallet test only. Right now essentially only those 30 lines of code is shared code, not the full 500.   If there is a desire to reference one unit test project from another, then we could just collapse the unit test projects into 1 and use folders for the separation and grouping of the tests per assembly. I raised an issue to discuss this,  becasue a single unit test project can be very effective.  It helps VS cope better too when you do that too.

Reply:
Guys, this is a tricky one... I'm not really sure why we would ever want to Nuget Tests.Common? But then again, Tests.Common should only contain Stratis agnostic code, if you want to purist about it (and if going to go down that routeO. I think for now the best think would just have to be to duplicate the wallet helper code out the to test assemblies that require them. That, or we have a Wallet.Tests.Common assembly but that seems like overkill. At the end of the day we can't have unit test assemblies referencing each other.

Reply:
My preference is collapsing the unit tests into a single unit test project. Then the wallet code can be used by any test that needs it. If we need to keep them separate for another reason, there is no real harm in duplicating the used code as per a couple of commits ago on this branch. Yes is breaks DRY but it is really quite minor considering the tests themselves are the protection layer around that code.  The idea of having a nugetted test common was really only about having the shared Stratis toolset that any test project can add, and wallettesthelper slipping in here in hindsight was a mistake.

Reply:
>My preference is collapsing the unit tests into a single unit test project.

I cannot imagine how we are going coordinate such an undertaking at this point with all the open PRs and multiple other projects (Smart Contracts / ICO / SideChains) tracking the FN master repo. So I don't think this will be possible at this time. But we still need to de-reference those test assemblies regardless. So whilst it not optimal, duplicating the wallet helper code to the various assemblies that require them, is at least a step in the right direction ;)

Reply:
Ah yes, fair enough. Ok I’ll do that then so we can proceed, later if we want to consolidate that code into something shared, it really wouldn’t be that hard.

Reply:
`So whilst it not optimal, duplicating the wallet helper code to the various assemblies that require them, is at least a step in the right direction ;)`
Nope, it's not.
Why can't we have tests projects referencing each other?
I don't want to see duplication of code, certainly when there are alternatives.